src/test/java/bio/terra/service/dataset/DatasetIntegrationTest.java

```diff
@@ -195,4 +216,118 @@ public void testAssetCreationUndo() throws Exception {
+        return new DataDeletionRequest()
+            .deleteType(DataDeletionRequest.DeleteTypeEnum.SOFT)
+            .specType(DataDeletionRequest.SpecTypeEnum.GCSFILE);
+    }
+
+    // TODO: wildcard test, validation tests, empty file, missing file, fault insertion on creating ext tables
+
+    private List<String> getRowIds(BigQuery bigQuery, DatasetModel dataset, String tableName, Long n) {
+        String tableRef = BigQueryFixtures.makeTableRef(dataset, tableName);
+        String sql = String.format("SELECT %s FROM %s LIMIT %s", PdaoConstant.PDAO_ROW_ID_COLUMN, tableRef, n);
+        TableResult result = BigQueryFixtures.query(sql, bigQuery);
+        assertThat("got right num of row ids back", result.getTotalRows(), equalTo(n));
+        return StreamSupport.stream(result.getValues().spliterator(), false)
+            .map(fieldValues -> fieldValues.get(0).getStringValue())
+            .collect(Collectors.toList());
+    }
+
+    private String writeListToScratch(String prefix, List<String> contents) throws IOException {
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        String targetPath = "scratch/" + prefix + "/" + UUID.randomUUID().toString() + ".csv";
+        BlobInfo blob = BlobInfo.newBuilder(testConfiguration.getIngestbucket(), targetPath).build();
+        try (WriteChannel writer = storage.writer(blob)) {
+            for (String line : contents) {
+                writer.write(ByteBuffer.wrap((line + "\n").getBytes(Charsets.UTF_8)));
+            }
+        }
+        return String.format("gs://%s/%s", blob.getBucket(), targetPath);
+    }
+
+    private void assertTableCount(BigQuery bigQuery, DatasetModel dataset, String tableName, Long n) {
+        String sql = "SELECT count(*) FROM " + BigQueryFixtures.makeTableRef(dataset, tableName);
+        TableResult result = BigQueryFixtures.query(sql, bigQuery);
+        assertThat("count matches", result.getValues().iterator().next().get(0).getLongValue(), equalTo(n));
+    }
+
+    public String ingestedDataset() throws Exception {
+        DatasetSummaryModel datasetSummaryModel = dataRepoFixtures.createDataset(steward(), "ingest-test-dataset.json");
+        String datasetId = datasetSummaryModel.getId();
+        IngestRequestModel ingestRequest = dataRepoFixtures.buildSimpleIngest(
+            "participant", "ingest-test/ingest-test-participant.json");
+        IngestResponseModel ingestResponse = dataRepoFixtures.ingestJsonData(steward(), datasetId, ingestRequest);
+        assertThat("correct participant row count", ingestResponse.getRowCount(), equalTo(5L));
+
+        ingestRequest = dataRepoFixtures.buildSimpleIngest(
+            "sample", "ingest-test/ingest-test-sample.json");
+        ingestResponse = dataRepoFixtures.ingestJsonData(steward(), datasetId, ingestRequest);
+        assertThat("correct sample row count", ingestResponse.getRowCount(), equalTo(7L));
+        return datasetId;
+    }
```

Comment:
I think variations of all these helpers existing in the `BigQueryPdao` connected test. Would anything here be easier / more reliable from testing at that level instead of at the full integration level? I guess the lines have already blurred pretty badly between the two kinds of tests...